src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -246,11 +246,14 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
         private static TagHelperAttributeDescriptor ToAttributeDescriptor(PropertyInfo property)
         {
             var attributeNameAttribute = property.GetCustomAttribute<HtmlAttributeNameAttribute>(inherit: false);
-            var attributeName = attributeNameAttribute != null ?
-                                attributeNameAttribute.Name :
-                                ToHtmlCase(property.Name);
-
-            return new TagHelperAttributeDescriptor(attributeName, property.Name, property.PropertyType.FullName);
+            var attributeName = attributeNameAttribute?.Name ?? ToHtmlCase(property.Name);
+            var isStringProperty = property.PropertyType == typeof(string);
+
+            return new TagHelperAttributeDescriptor(
+                attributeName,
+                property.Name,
+                property.PropertyType.FullName,
+                isStringProperty);
```

Comment:
Why put the `isStringProperty` on the descriptor when it can be mostly derived from the type name? This seems like an unrelated change and is adding unnecessary data to the should be dumb `TagHelperAttributeDescriptor`